card.go

```diff
@@ -170,3 +185,24 @@ func deepClone(ix interface{}) interface{} {
 	}
 	return ix
 }
+
+func fillNameComponentMap(m map[string]interface{}, items interface{}) {
+	for _, v := range items.([]interface{}) {
+		// This map always has a single key - wrapper so this is O(1) not O(n).
+		for _, v := range v.(map[string]interface{}) {
+			component := v.(map[string]interface{})
+			if name, ok := component["name"]; ok {
+				m[name.(string)] = v
```

Comment:
To be safe here, I'd do a `type switch` or just check the correctness of type with `if`:
```go
if n, ok := name.(string); ok {
    m[n] = v
}
```

Reply:
Will do.